### ISSUE TITLE
Loading/Initialization of ADVR Joint/Command msgs gives warning/error

### ISSUE THREAD

**Issue by liesrock** [ADVRHumanoids/XCM#42]
devel branch.

As title: then the CommunicationHandler works as expected.
Tested in simulation (14.04) with both CENTAURO - WALK-MAN.

Looks like the call stack leads to the initialization of the TX msg first (because of the getRobot in the RX), that expect some data structure initialized by the RX: this brings us to the WARNING/ERROR in initialization.

```
ERROR No message received on topic /xbotcore/centauro/joint_states! Read() won't work!!
WARNING in getIndex! Joint hip_yaw_1 is not defined inside the ROS joint state message!
ERROR joint hip_yaw_1 does not exist within joint state!
WARNING in getIndex! Joint hip_pitch_1 is not defined inside the ROS joint state message!
ERROR joint hip_pitch_1 does not exist within joint state!
WARNING in getIndex! Joint knee_pitch_1 is not defined inside the ROS joint state message!
ERROR joint knee_pitch_1 does not exist within joint state!
WARNING in getIndex! Joint ankle_pitch_1 is not defined inside the ROS joint state message!
ERROR joint ankle_pitch_1 does not exist within joint state!
WARNING in getIndex! Joint ankle_yaw_1 is not defined inside the ROS joint state message!
ERROR joint ankle_yaw_1 does not exist within joint state!
WARNING in getIndex! Joint j_wheel_1 is not defined inside the ROS joint state message!
ERROR joint j_wheel_1 does not exist within joint state!
WARNING in getIndex! Joint hip_yaw_2 is not defined inside the ROS joint state message!
ERROR joint hip_yaw_2 does not exist within joint state!
WARNING in getIndex! Joint hip_pitch_2 is not defined inside the ROS joint state message!
ERROR joint hip_pitch_2 does not exist within joint state!
WARNING in getIndex! Joint knee_pitch_2 is not defined inside the ROS joint state message!
ERROR joint knee_pitch_2 does not exist within joint state!
WARNING in getIndex! Joint ankle_pitch_2 is not defined inside the ROS joint state message!
ERROR joint ankle_pitch_2 does not exist within joint state!
WARNING in getIndex! Joint ankle_yaw_2 is not defined inside the ROS joint state message!
ERROR joint ankle_yaw_2 does not exist within joint state!
WARNING in getIndex! Joint j_wheel_2 is not defined inside the ROS joint state message!
ERROR joint j_wheel_2 does not exist within joint state!
WARNING in getIndex! Joint j_arm1_1 is not defined inside the ROS joint state message!
ERROR joint j_arm1_1 does not exist within joint state!
WARNING in getIndex! Joint j_arm1_2 is not defined inside the ROS joint state message!
ERROR joint j_arm1_2 does not exist within joint state!
WARNING in getIndex! Joint j_arm1_3 is not defined inside the ROS joint state message!
ERROR joint j_arm1_3 does not exist within joint state!
WARNING in getIndex! Joint j_arm1_4 is not defined inside the ROS joint state message!
ERROR joint j_arm1_4 does not exist within joint state!
WARNING in getIndex! Joint j_arm1_5 is not defined inside the ROS joint state message!
ERROR joint j_arm1_5 does not exist within joint state!
WARNING in getIndex! Joint j_arm1_6 is not defined inside the ROS joint state message!
ERROR joint j_arm1_6 does not exist within joint state!
WARNING in getIndex! Joint j_arm1_7 is not defined inside the ROS joint state message!
ERROR joint j_arm1_7 does not exist within joint state!
WARNING in getIndex! Joint neck_yaw is not defined inside the ROS joint state message!
ERROR joint neck_yaw does not exist within joint state!
WARNING in getIndex! Joint neck_pitch is not defined inside the ROS joint state message!
ERROR joint neck_pitch does not exist within joint state!
WARNING in getIndex! Joint hip_yaw_3 is not defined inside the ROS joint state message!
ERROR joint hip_yaw_3 does not exist within joint state!
WARNING in getIndex! Joint hip_pitch_3 is not defined inside the ROS joint state message!
ERROR joint hip_pitch_3 does not exist within joint state!
WARNING in getIndex! Joint knee_pitch_3 is not defined inside the ROS joint state message!
ERROR joint knee_pitch_3 does not exist within joint state!
WARNING in getIndex! Joint ankle_pitch_3 is not defined inside the ROS joint state message!
ERROR joint ankle_pitch_3 does not exist within joint state!
WARNING in getIndex! Joint ankle_yaw_3 is not defined inside the ROS joint state message!
ERROR joint ankle_yaw_3 does not exist within joint state!
WARNING in getIndex! Joint j_wheel_3 is not defined inside the ROS joint state message!
ERROR joint j_wheel_3 does not exist within joint state!
WARNING in getIndex! Joint hip_yaw_4 is not defined inside the ROS joint state message!
ERROR joint hip_yaw_4 does not exist within joint state!
WARNING in getIndex! Joint hip_pitch_4 is not defined inside the ROS joint state message!
ERROR joint hip_pitch_4 does not exist within joint state!
WARNING in getIndex! Joint knee_pitch_4 is not defined inside the ROS joint state message!
ERROR joint knee_pitch_4 does not exist within joint state!
WARNING in getIndex! Joint ankle_pitch_4 is not defined inside the ROS joint state message!
ERROR joint ankle_pitch_4 does not exist within joint state!
WARNING in getIndex! Joint ankle_yaw_4 is not defined inside the ROS joint state message!
ERROR joint ankle_yaw_4 does not exist within joint state!
WARNING in getIndex! Joint j_wheel_4 is not defined inside the ROS joint state message!
ERROR joint j_wheel_4 does not exist within joint state!
WARNING in getIndex! Joint j_arm2_1 is not defined inside the ROS joint state message!
ERROR joint j_arm2_1 does not exist within joint state!
WARNING in getIndex! Joint j_arm2_2 is not defined inside the ROS joint state message!
ERROR joint j_arm2_2 does not exist within joint state!
WARNING in getIndex! Joint j_arm2_3 is not defined inside the ROS joint state message!
ERROR joint j_arm2_3 does not exist within joint state!
WARNING in getIndex! Joint j_arm2_4 is not defined inside the ROS joint state message!
ERROR joint j_arm2_4 does not exist within joint state!
WARNING in getIndex! Joint j_arm2_5 is not defined inside the ROS joint state message!
ERROR joint j_arm2_5 does not exist within joint state!
WARNING in getIndex! Joint j_arm2_6 is not defined inside the ROS joint state message!
ERROR joint j_arm2_6 does not exist within joint state!
WARNING in getIndex! Joint j_arm2_7 is not defined inside the ROS joint state message!
ERROR joint j_arm2_7 does not exist within joint state!
WARNING in getIndex! Joint torso_yaw is not defined inside the ROS joint state message!
ERROR joint torso_yaw does not exist within joint state!
WARNING in getIndex! Joint neck_velodyne is not defined inside the ROS joint state message!
ERROR joint neck_velodyne does not exist within joint state!
WARNING in getIndex! Joint hip_yaw_1 is not defined inside the ROS controller!
ERROR joint hip_yaw_1 does not exist within controller!
WARNING in getIndex! Joint hip_pitch_1 is not defined inside the ROS controller!
ERROR joint hip_pitch_1 does not exist within controller!
WARNING in getIndex! Joint knee_pitch_1 is not defined inside the ROS controller!
ERROR joint knee_pitch_1 does not exist within controller!
WARNING in getIndex! Joint ankle_pitch_1 is not defined inside the ROS controller!
ERROR joint ankle_pitch_1 does not exist within controller!
WARNING in getIndex! Joint ankle_yaw_1 is not defined inside the ROS controller!
ERROR joint ankle_yaw_1 does not exist within controller!
WARNING in getIndex! Joint j_wheel_1 is not defined inside the ROS controller!
ERROR joint j_wheel_1 does not exist within controller!
WARNING in getIndex! Joint hip_yaw_2 is not defined inside the ROS controller!
ERROR joint hip_yaw_2 does not exist within controller!
WARNING in getIndex! Joint hip_pitch_2 is not defined inside the ROS controller!
ERROR joint hip_pitch_2 does not exist within controller!
WARNING in getIndex! Joint knee_pitch_2 is not defined inside the ROS controller!
ERROR joint knee_pitch_2 does not exist within controller!
WARNING in getIndex! Joint ankle_pitch_2 is not defined inside the ROS controller!
ERROR joint ankle_pitch_2 does not exist within controller!
WARNING in getIndex! Joint ankle_yaw_2 is not defined inside the ROS controller!
ERROR joint ankle_yaw_2 does not exist within controller!
WARNING in getIndex! Joint j_wheel_2 is not defined inside the ROS controller!
ERROR joint j_wheel_2 does not exist within controller!
WARNING in getIndex! Joint j_arm1_1 is not defined inside the ROS controller!
ERROR joint j_arm1_1 does not exist within controller!
WARNING in getIndex! Joint j_arm1_2 is not defined inside the ROS controller!
ERROR joint j_arm1_2 does not exist within controller!
WARNING in getIndex! Joint j_arm1_3 is not defined inside the ROS controller!
ERROR joint j_arm1_3 does not exist within controller!
WARNING in getIndex! Joint j_arm1_4 is not defined inside the ROS controller!
ERROR joint j_arm1_4 does not exist within controller!
WARNING in getIndex! Joint j_arm1_5 is not defined inside the ROS controller!
ERROR joint j_arm1_5 does not exist within controller!
WARNING in getIndex! Joint j_arm1_6 is not defined inside the ROS controller!
ERROR joint j_arm1_6 does not exist within controller!
WARNING in getIndex! Joint j_arm1_7 is not defined inside the ROS controller!
ERROR joint j_arm1_7 does not exist within controller!
WARNING in getIndex! Joint neck_yaw is not defined inside the ROS controller!
ERROR joint neck_yaw does not exist within controller!
WARNING in getIndex! Joint neck_pitch is not defined inside the ROS controller!
ERROR joint neck_pitch does not exist within controller!
WARNING in getIndex! Joint hip_yaw_3 is not defined inside the ROS controller!
ERROR joint hip_yaw_3 does not exist within controller!
WARNING in getIndex! Joint hip_pitch_3 is not defined inside the ROS controller!
ERROR joint hip_pitch_3 does not exist within controller!
WARNING in getIndex! Joint knee_pitch_3 is not defined inside the ROS controller!
ERROR joint knee_pitch_3 does not exist within controller!
WARNING in getIndex! Joint ankle_pitch_3 is not defined inside the ROS controller!
ERROR joint ankle_pitch_3 does not exist within controller!
WARNING in getIndex! Joint ankle_yaw_3 is not defined inside the ROS controller!
ERROR joint ankle_yaw_3 does not exist within controller!
WARNING in getIndex! Joint j_wheel_3 is not defined inside the ROS controller!
ERROR joint j_wheel_3 does not exist within controller!
WARNING in getIndex! Joint hip_yaw_4 is not defined inside the ROS controller!
ERROR joint hip_yaw_4 does not exist within controller!
WARNING in getIndex! Joint hip_pitch_4 is not defined inside the ROS controller!
ERROR joint hip_pitch_4 does not exist within controller!
WARNING in getIndex! Joint knee_pitch_4 is not defined inside the ROS controller!
ERROR joint knee_pitch_4 does not exist within controller!
WARNING in getIndex! Joint ankle_pitch_4 is not defined inside the ROS controller!
ERROR joint ankle_pitch_4 does not exist within controller!
WARNING in getIndex! Joint ankle_yaw_4 is not defined inside the ROS controller!
ERROR joint ankle_yaw_4 does not exist within controller!
WARNING in getIndex! Joint j_wheel_4 is not defined inside the ROS controller!
ERROR joint j_wheel_4 does not exist within controller!
WARNING in getIndex! Joint j_arm2_1 is not defined inside the ROS controller!
ERROR joint j_arm2_1 does not exist within controller!
WARNING in getIndex! Joint j_arm2_2 is not defined inside the ROS controller!
ERROR joint j_arm2_2 does not exist within controller!
WARNING in getIndex! Joint j_arm2_3 is not defined inside the ROS controller!
ERROR joint j_arm2_3 does not exist within controller!
WARNING in getIndex! Joint j_arm2_4 is not defined inside the ROS controller!
ERROR joint j_arm2_4 does not exist within controller!
WARNING in getIndex! Joint j_arm2_5 is not defined inside the ROS controller!
ERROR joint j_arm2_5 does not exist within controller!
WARNING in getIndex! Joint j_arm2_6 is not defined inside the ROS controller!
ERROR joint j_arm2_6 does not exist within controller!
WARNING in getIndex! Joint j_arm2_7 is not defined inside the ROS controller!
ERROR joint j_arm2_7 does not exist within controller!
WARNING in getIndex! Joint torso_yaw is not defined inside the ROS controller!
ERROR joint torso_yaw does not exist within controller!
WARNING in getIndex! Joint neck_velodyne is not defined inside the ROS controller!
ERROR joint neck_velodyne does not exist within controller!
```

**Comment by alaurenzi** [comment]
Ok this should not create issues but I will solve it since is not nice to read

**Comment by liesrock** [comment]
Yes, it is not a bug sorry.

**Comment by liesrock** [comment]
Solved by @alaurenzi in cd70645
